### ISSUE TITLE
Support blank fsType on NFS volumes

### ISSUE THREAD

**Issue by sbarre-esit** [NetApp/trident#402]
**Describe the bug**
When k8s mounts a volume, it will attempt to set the group ownership on the files in the volume. https://github.com/kubernetes/kubernetes/blob/725635e/pkg/volume/csi/csi_mounter.go#L388-L393

If the volume has a fsType and is RWO it will trigger and try and chown all the files in the volume. https://github.com/kubernetes/kubernetes/blob/725635e/pkg/volume/volume_linux.go#L71

Currently the CSI sidecar defaults the fsType to `ext4` and Trident doesn't overwrite this. So when mounting a RWO NFS PVC this code is called. With certain permissions on the volume mount path and subfolders this can then fail.

https://github.com/kubernetes-csi/external-provisioner/pull/400 is tracking a change to CSI that will change the default fsType to blank. Once that is done the new sidecar needs to get added into Trident. This issue is to track when that gets done.

**Environment**

- Trident version: 20.01.0
- Trident installation flags used: -n trident
- Container runtime: cri-o-1.16.6-11.dev.rhaos4.3.gitdf26a5a.el8.x86_64
- Kubernetes version: v1.16.2
- Kubernetes orchestrator: OpenShift v4.3.18
- Kubernetes enabled feature gates: None/Default
- OS: RHCOS
- NetApp backend types: ONTAP Select
- Other:

**To Reproduce**
Build a docker image

```docker
FROM registry.access.redhat.com/ubi7
RUN mkdir /foo && chown 26:0 /foo && chmod 770 /foo
USER 26
```

Create a PVC using the ontap-nas-economy driver that is RWO. Create a pod that mounts that PVC to `/foo`. `rsh` into the pod and run

```console
sh# mkdir /foo/folder1
sh# chmod 700 /foo/folder1
```

Restart the pod and it will fail to come back up saying something like this in the event log/ pod description.

```
MountVolume.SetUp failed for volume "pvc-74831e94-1426-42b6-85fa-7fcb217844ac" : applyFSGroup failed for vol pvc-74831e94-1426-42b6-85fa-7fcb217844ac: open /var/lib/kubelet/pods/09d74aaa-728c-4a7a-91f6-8a28ae998a00/volumes/kubernetes.io~csi/pvc-74831e94-1426-42b6-85fa-7fcb217844ac/mount/foo: permission denied
```

**Expected behavior**
Don't run `applyFSGroup()` at all

**Workarounds**

When Trident provisions a qtree volume it assigns the vservers `default` export policy to the qtree instead of the `trident_qtree_pool_export_policy` one.

So our options to work around the `applyFSGroups` issue is to

1. Set default export policy to Superuser Security Type = Any

   ```
   SVM1::> vserver export-policy rule modify  -policyname default -ruleindex 1 -superuser any
   ```

2. Tell Trident to use the trident created policy for qtrees

   Update the backend.json with

   ```json
    "defaults": { "exportPolicy": "trident_qtree_pool_export_policy"}
   ```

3. Set the default export policy User ID To Which Anonymous Users Are Mapped to zero or root

   ```
   SVM1::> vserver export-policy rule modify  -policyname default -ruleindex 1 -anon 0
   ```

**Comment by bigg01** [comment]
We are facing this problem here too in Openshift 4. overwritting permission is realy ugly.  Can we tell the csi privisoner not to so ? Is there a flag not apply applyFSGroup or can we configure a specfic one? Its really  not usable like that. BG

**Comment by gnarl** [comment]
A [fix](https://github.com/kubernetes-csi/external-provisioner/pull/400) was added in the CSI external-provisioner which is to be included in the CSI Provisioner v2.0.0 release.

**Comment by bigg01** [comment]
Thank you for replay. Do you think it will be included in 20.07?

**Comment by gnarl** [comment]
Hi @bigg01, 

We don't control when the CSI sidecar versions are released. If the csi-provisioner v2.0.0 is released in time for the Trident 20.07.0 release we will update our yaml definitions. If it doesn't make the Trident release then you still have the option to generate the yaml needed for installing Trident and updating the version of the sidecar for your installation. We will be able to provide instructions on how to do this if this is the case.

**Comment by gnarl** [comment]
The cdi-provisioner sidecar v2.0.1 is included in the deployment for the [Trident v20.07.1 release](https://github.com/NetApp/trident/releases/tag/v20.07.1).

**Comment by sbarre-esit** [comment]
Protip: After this fix you have to make sure you set a `fsType` on your block volume storage classes, else it won't chgrp on them and then you can't write to them ....
